pkg/component/watch_test.go

```diff
@@ -55,3 +67,291 @@ func TestIsRegExpMatch(t *testing.T) {
 	}
 
 }
+
+func setUpFabric8AnalyticsLicenseAnalysisComponentSrc(componentName string, requiredFilePaths []FileProperties) (string, map[string]FileProperties, error) {
+	// Setup component dir structure
+	retVal := make(map[string]FileProperties)
+	dirTreeMappings := make(map[string]string)
+	currentUser, err := user.Current()
+	if err != nil {
+		return "", retVal, errors.Wrapf(err, "failed to get current user home directory")
+	}
+	basePath := currentUser.HomeDir
```

Comment:
Add comments, what is this section doing?

Reply:
Ack

Reply:
that means that the test is writing to users home?? 
I don't want tests to write to my home

Reply:
It'll be cleared as well...
Would /tmp be ok?

Reply:
Can't we use some kind of temp dir?

Reply:
@geoand That's what is being done but the temp dir needs to have a parent and the problem is having a cross-platform way to select a parent… i.e. not sure writing to use `/tmp` as a parent would work on Windows…

Reply:
I'm also concerned about the temp dir being deleted in all situations (i.e. if there's an app crash that happens before we explicitly delete it).

Reply:
Agreed, but I don't think having a reference to the home dir is a good idea in case something goes wrong.

Futhermore, `odo` as a binary is supposed to work on Windoss, but is development on Windows part of that support?
I mean this is just a test and if `odo` is meant to be developed on Mac or Linux, then there is no issue :)

Of course a totally OS independent implementation would be ideal

Reply:
> I'm also concerned about the temp dir being deleted in all situations (i.e. if there's an app crash that happens before we explicitly delete it).

@metacosm Its being deleted as soon as  the function returns via https://github.com/redhat-developer/odo/pull/820/files#diff-dd5e8748b905a08c9dbc3ef0de929481R282 Basically, the defer there ensures that the cleanup function completes immediately before the surrounding function returns(even if it is for error)

Reply:
> Agreed, but I don't think having a reference to the home dir is a good idea in case something goes wrong.
> 
> Futhermore, `odo` as a binary is supposed to work on Windos, but is development on Windows part of that support?
> I mean this is just a test and if `odo` is meant to be developed on Mac or Linux, then there is no issue :)
> 
> Of course a totally OS independent implementation would be ideal

@geoand @metacosm Yes I am looking for a platform independent way of performing file CRUD operations so `/tmp` might not be relavant to windows(windows paths start with volume name)...
And yes our tests also run on appveyor windows env(for the moment only unit tests), they are also expected to be platform independent(some of our travis jobs also run on mac os)

Reply:
Aha that seems to be working as well :)